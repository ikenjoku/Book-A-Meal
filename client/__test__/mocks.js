
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
