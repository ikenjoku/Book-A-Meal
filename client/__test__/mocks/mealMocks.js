export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
