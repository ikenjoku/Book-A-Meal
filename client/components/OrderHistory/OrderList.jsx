import React, { Component } from 'react';
import { connect } from 'react-redux';
import Loader from 'react-loader-spinner';
import { deliverAnOrder } from "../../actions/orderActions";

const OrderList = ({ orders, deliverAnOrder, isLoadingOrders }) => {
  let daysTotal = 0;
  let serialNum = 0
  orders.map(order =>{ if(order.status === 'delivered')  daysTotal += order.amount});
  const renderedJsx = orders.length === 0 ? 
  <p className='error-alert'>No Orders found for this day</p> :
  <div>
  <table className="order-table-body">
        <thead>
          <tr>
            <th>S/N</th>
            <th>Customer</th>
            <th>Order Summary</th>
            <th>Amount</th>
            <th>Status</th>
            <th>Action</th>
          </tr>
        </thead>
          <tbody>
          { orders.map(order =>
            <tr key={order.id}>
              <td >{++serialNum}</td>
              <td>{order.User.firstname} {order.User.lastname}</td>
              <td>{order.Meal.name}</td>
              <td>&#8358;{order.amount}</td>
              <td>{order.status}</td>
              <td>{order.status === 'pending'  && <button className='deliverBtn' onClick={() => deliverAnOrder(order.id)}>deliver</button>}</td>
            </tr>
            )}
          </tbody>          
      </table>
      <div id="total-made">
        <h2>Total: &#8358;{daysTotal}</h2>
      </div>
      </div>
  
  return (
    <div>
      {
        isLoadingOrders ? 
        <div className='loading-spinner'>
          <Loader 
              type="Circles"
              color="#9D2401"
              height="100"
              width="100"
          />
          <h3 id='loader-text'>Fetching...</h3>  
          </div> : renderedJsx
      }
    </div>
  );
}

const mapStateToProps = state => {
  return{
    orders: state.orderReducer.orders,
    isLoadingOrders: state.orderReducer.isLoadingOrders,
    error: state.orderReducer.error
  }
}
export default connect(mapStateToProps, { deliverAnOrder })(OrderList);