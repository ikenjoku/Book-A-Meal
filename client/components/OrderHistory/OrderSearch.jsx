import React, { Component } from 'react';


export default class OrderSearch extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="order-history-date-picker center">
        <form>
          <input
            className="input-field-control"
            type="date"
            placeholder="Choose a date..."
            name="history-date" />
          <input
            className="input-btn-control"
            type="submit"
            value="Choose date" />
        </form>
      </div>
    );
  }
}
