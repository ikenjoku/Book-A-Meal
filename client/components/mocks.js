

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

