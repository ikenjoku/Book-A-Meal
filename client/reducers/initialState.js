export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
