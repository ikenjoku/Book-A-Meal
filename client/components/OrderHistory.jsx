import React, { Component } from 'react';

import NavBar from './NavBar.jsx';

import { sampleOrders } from "./mocks";

export default class OrderHistory extends Component{
  constructor(props){
    super(props);
    this.state = {
      orders: sampleOrders.orders
    }
  }


  render(){
    return (
      <div className="contain-page">
      <NavBar/>
      <main id='order-history-container' className="order-history-content">
          <div className="cool-lg-text">
              <h2>Order History</h2>
          </div>
          <OrderSearch/>
          <OrderList orders={this.state.orders}/>
      </main>
      <footer className="footer order-history-footer">
          <p>Book-A-Meal &copy;2018</p>
      </footer>
  </div>
    );
  }
}


class OrderSearch extends Component{
  constructor(props){
    super(props);
  }

  render(){
    return (
      <div className="order-history-date-picker center">
        <form>
          <input 
            className="input-field-control" 
            type="date" 
            placeholder="Choose a date..."  
            name="history-date" />
          <input 
            className="input-btn-control" 
            type="submit" 
            value="Choose date" />
        </form>
    </div>
    );
  }
}

const OrderList = ({orders}) => {

  return (
    <div>
      <div className="order-table-header">
        <p>Order ID</p>
        <p>Customer</p>
        <p>Order Summary</p>
        <p>Amount</p>
      </div>
      <div className="order-table-body">
        { orders.map(order =>
          <div key={order.id} className="order-table-body-row">
              <p>{order.id}</p>
              <p>{order.User.firstname} {order.User.lastname}</p>
              <div className="order-tbody-summary">
                <p>{order.Meal.name}</p>
              </div>
              <p>&#8358;{order.amount}</p>
          </div>
        ) }
      </div>
      <div id="total-made">
          <h2>Total: &#8358;150 470.00</h2>
      </div>
  </div>
  );
}