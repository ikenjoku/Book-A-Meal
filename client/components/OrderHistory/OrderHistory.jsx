import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import DatePicker from '../DatePicker.jsx';
import OrderList from './OrderList.jsx';
import { getOrdersByDate } from "../../actions/orderActions";


class OrderHistory extends Component {
  constructor(props) {
    super(props);

  }
  componentDidMount() {
    const today = new Date().toISOString().substr(0, 10);
    this.props.getOrdersByDate({ selectedDate: today });
  }

  render() {
    return (
      <main id='order-history-container' className="order-history-content">
        <div className="cool-lg-text">
          <h2>Order History</h2>
        </div>
        <DatePicker btnName='Get Orders' onSubmit={
          (selectedDate) => this.props.getOrdersByDate(selectedDate)
        } />
        <OrderList />
      </main>
    );
  }
}

export default connect(null, { getOrdersByDate })(OrderHistory);
