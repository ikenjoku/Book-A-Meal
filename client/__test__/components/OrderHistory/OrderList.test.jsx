import React from 'react';
import { shallow } from 'enzyme';
import { orders } from '../../mocks';
import { OrderList, mapStateToProps } from '../../../components/OrderHistory/OrderList';
import initialState from '../../../reducers/initialState';

describe('OrderList', () => {
  let wrapper;
  const deliverAnOrder = jest.fn();
  const isLoadingOrders = false;

  test('should render the OrderList correctly', () => {
    wrapper = shallow(<OrderList
      deliverAnOrder={deliverAnOrder}
      isLoadingOrders={isLoadingOrders}
      orders={orders}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render a notice when there are no orders', () => {
    wrapper = shallow(<OrderList
      deliverAnOrder={deliverAnOrder}
      isLoadingOrders={isLoadingOrders}
      orders={[]}
    />);
    expect(wrapper.find('.error-alert').length).toBe(1);
  });

  test('should render a loader still loading orders', () => {
    wrapper = shallow(<OrderList
      deliverAnOrder={deliverAnOrder}
      isLoadingOrders={true}
      orders={orders}
    />);
    expect(wrapper.find('Loader').length).toBe(1);
  });

  test('should map state to props', () => {
    const ownProps = { orders, isLoadingOrders: false };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
