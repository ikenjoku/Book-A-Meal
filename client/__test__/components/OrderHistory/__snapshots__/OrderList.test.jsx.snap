// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList should map state to props 1`] = `
Object {
  "isLoadingOrders": false,
  "orders": Array [],
}
`;

exports[`OrderList should render the OrderList correctly 1`] = `
<div>
  <div>
    <table
      className="order-table-body"
    >
      <thead>
        <tr>
          <th>
            S/N
          </th>
          <th>
            Customer
          </th>
          <th>
            Order Summary
          </th>
          <th>
            Amount
          </th>
          <th>
            Status
          </th>
          <th>
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="2"
        >
          <td>
            1
          </td>
          <td>
            Ezinne
             
            Njoku
          </td>
          <td>
            Rice and Beans
          </td>
          <td>
            ₦
            1500
          </td>
          <td>
            cancelled
          </td>
          <td />
        </tr>
        <tr
          key="3"
        >
          <td>
            2
          </td>
          <td>
            Kelechi
             
            Njoku
          </td>
          <td>
            Egusi Semo
          </td>
          <td>
            ₦
            1200
          </td>
          <td>
            pending
          </td>
          <td>
            <button
              className="deliverBtn"
              id="deliver-order-3"
              onClick={[Function]}
            >
              deliver
            </button>
          </td>
        </tr>
        <tr
          key="4"
        >
          <td>
            3
          </td>
          <td>
            Nnamdi
             
            James
          </td>
          <td>
            Egusi Semo
          </td>
          <td>
            ₦
            1500
          </td>
          <td>
            delivered
          </td>
          <td />
        </tr>
      </tbody>
    </table>
    <div
      id="total-made"
    >
      <h2>
        Total: ₦
        1500
      </h2>
    </div>
  </div>
</div>
`;
